nat-lab/tests/conftest.py

```diff
@@ -86,39 +90,76 @@ async def on_output(output: str) -> None:
             "tcp",
             "start=" + str(start_port),
             "num=" + str(num_ports),
-        ]).execute(on_output, on_output)
+        ]
+    elif vm_tag is ConnectionTag.MAC_VM:
+        cmd = [
+            "sysctl",
+            "-w",
+            "net.inet.ip.portrange.first=" + str(start_port),
+            "net.inet.ip.portrange.last=" + str(start_port + num_ports),
+        ]
+    else:
+        # Linux
+        cmd = [
+            "sysctl",
+            "-w",
+            "net.ipv4.ip_local_port_range="
+            + str(start_port)
+            + " "
+            + str(start_port + num_ports),
+        ]
+
+    async with new_connection_raw(vm_tag) as connection:
+        await connection.create_process(cmd).execute(on_output, on_output)
 
 
 @pytest.fixture(autouse=True)
-@pytest.mark.windows
-def setup_windows_ports(request):
+def setup_ephemeral_ports(request):
     test_name = request.node.name
 
     def execute_setup(vm_tag):
         try:
-            asyncio.run(win_ports(vm_tag))
+            asyncio.run(os_ephemeral_ports(vm_tag))
         except ProcessExecError as e:
-            print(f"win_ports_{vm_tag} process execution failed: {e}")
+            print(
+                datetime.now(),
+                f"os_ephemeral_ports_{vm_tag} process execution failed: {e}",
+            )
 
-    if "[WINDOWS_VM_" in test_name:
-        # Extract all VM numbers (WINDOWS_VM_1 and/or WINDOWS_VM_2)
+    # Handle parameterized test names
+    if (
+        "[" in test_name
+        and "]" in test_name
+        and (("DOCKER_" in test_name and "CLIENT" in test_name) or "_VM_" in test_name)
+    ):
+        # Extract all connection tags from test name
         vm_names = [
             param
             for param in test_name.split("[")[1].split("]")[0].split("-")
-            if param.startswith("WINDOWS_VM_")
+            if param in ConnectionTag.__members__
         ]
 
         for vm_name in vm_names:
             execute_setup(ConnectionTag[vm_name])
     else:
+        # Handle test markers
         items = request.session.items
         for item in items:
             for mark in item.own_markers:
-                if mark.name != "windows":
-                    continue
-                for tag in [ConnectionTag.WINDOWS_VM_1, ConnectionTag.WINDOWS_VM_2]:
-                    execute_setup(tag)
-
+                if mark.name == "windows":
+                    for tag in [ConnectionTag.WINDOWS_VM_1, ConnectionTag.WINDOWS_VM_2]:
+                        execute_setup(tag)
+                elif mark.name == "mac":
+                    execute_setup(ConnectionTag.MAC_VM)
+                else:
```

Comment:
idk how to proceed with this, but if the test marker is windows or mac, doesn't mean that it doesn't contain docker container in it. I think this always has to be executed

Reply:
I fixed that by not executing every-time, but putting everything to a set.